src/charm.py

```diff
@@ -312,61 +313,118 @@ def _build_authorization_policies(self, mesh_info: List[MeshPolicy]):
+                            }
+                        ),
+                        rules=[
+                            Rule(
+                                from_=[  # type: ignore # this is accessible via an alias
+                                    From(
+                                        source=Source(
+                                            principals=[
+                                                _get_peer_identity_for_juju_application(
+                                                    policy.source_app_name, policy.source_namespace
+                                                )
+                                            ]
+                                        )
                                     )
-                                )
-                            ],
-                            to=[
-                                To(
-                                    operation=Operation(
-                                        # TODO: Make these ports strings instead of ints in endpoint?
-                                        ports=[str(p) for p in endpoint.ports]
-                                        if endpoint.ports
-                                        else [],
-                                        hosts=endpoint.hosts,
-                                        methods=endpoint.methods,
-                                        paths=endpoint.paths,
+                                ],
+                                to=[
+                                    To(
+                                        operation=Operation(
+                                            # TODO: Make these ports strings instead of ints in endpoint?
+                                            ports=[str(p) for p in endpoint.ports]
+                                            if endpoint.ports
+                                            else [],
+                                        )
                                     )
-                                )
-                                for endpoint in policy.endpoints
-                            ],
-                        )
-                    ],
-                    # by_alias=True because the model includes an alias for the `from` field
-                    # exclude_unset=True because unset fields will be treated as their default values in Kubernetes
-                    # exclude_none=True because null values in this data always mean the Kubernetes default
-                ).model_dump(by_alias=True, exclude_unset=True, exclude_none=True),
-            )
+                                    for endpoint in policy.endpoints
+                                ],
+                            ),
+                        ],
+                    ).model_dump(by_alias=True, exclude_unset=True, exclude_none=True),
+                )
+
+            # L7 policy created for target Juju applications (services)
+            else:
```

Comment:
This suggests that all other types are accepted, but really I think we only want .app here and to reject anything else.  wdyt?

```suggestion
            elif(policy.target_type == PolicyTargetType.app):
              ...
            else:
              raise
```